### PR TITLE
Enable defining headers for kafka2 and rdkafka2 output

### PR DESCRIPTION
**Pull request by PhilGrayson**
I'd like to be able to define headers on a Kafka message based on the content of a fluentd record.

This is due to the Splunk Kafka Connect plugin using Kafka headers to populate some metadata about an event in Splunk ([source](https://github.com/splunk/kafka-connect-splunk#headers-parameters)).

---

I've raised this PR to see if the maintainer is interested in the contribution. No offence taken if not, I'd be wary about accepting new code without any automated tests.

### CODE REVIEW

**Comment by PhilGrayson** [comment]
Headers should be an array, not a Hash. Kafka allows a record to have headers with duplicate keys.

File: lib/fluent/plugin/out_kafka2.rb [hunk 1]
```diff
@@ -39,6 +39,8 @@ class Fluent::Kafka2Output < Output
                  :desc => 'Set true to remove partition key from data'
     config_param :exclude_topic_key, :bool, :default => false,
                  :desc => 'Set true to remove topic name key from data'
+    config_param :headers, :hash, default: {}, symbolize_keys: true, value_type: :string
```
Comment (repeatedly) [inline lib/fluent/plugin/out_kafka2.rb]: need `:desc` paramaeter.

File: lib/fluent/plugin/out_kafka2.rb [hunk 2]
```diff
@@ -39,6 +39,8 @@ class Fluent::Kafka2Output < Output
...
+    config_param :headers_from_record, :hash, default: {}, symbolize_keys: true, value_type: :string
```
Comment (repeatedly) [inline lib/fluent/plugin/out_kafka2.rb]: ditto

**Comment by PhilGrayson** [comment]
Thanks for the review! I've pushed a new commit.

I've considered adding this feature to the 'kafka' and 'kafka_buffered' output. However it'll require a bump to the minimum fluentd version dependency. Do you have any thoughts on this?

Current gemspec has
```
gem.add_dependency "fluentd", [">= 0.10.58", "< 2"]
```
The handy 'record_accessor' helper was added in fluentd v0.14.20

**Comment by PhilGrayson** [comment]
> Headers should be an array, not a Hash. Kafka allows a record to have headers with duplicate keys.

Turns out [ruby-kafka](https://github.com/zendesk/ruby-kafka/blob/dce5db4d6ef7a8ec6dd0dc231df3a4029ecde05b/lib/kafka/producer.rb#L182) and [rdafka-ruby](https://github.com/appsignal/rdkafka-ruby/blob/10fe675583a0f33866357e79ddaf83025bb1f566/lib/rdkafka/producer.rb#L60]) accepts headers as a Hash only.

**Comment by repeatedly** [comment]
> I've considered adding this feature to the 'kafka' and 'kafka_buffered' output. However it'll require a bump to the minimum fluentd version dependency. Do you have any thoughts on this?

I think no need support for old API plugins. Copy `record_accessor` code is overenginnering for me.

**Comment by repeatedly** [comment]
Thanks!

**Pull request merged**